### PR TITLE
fix: serialization error in _vertex_to_primitive_dict for nested objects

### PR DESCRIPTION
**Pull request by dhlidongming**
This pull request resolves a serialization error encountered during flow execution when executing `log_transaction`.
https://github.com/langflow-ai/langflow/blob/c7fed78181a2a133917c1205c6fb3cdcf7d33559/src/backend/base/langflow/graph/utils.py#L149-L161
The issue was due to nested dict/list structures containing non-python types within components' inputs, for example, components with `DictInput` type inputs.

The solution involves modifying the `_vertex_to_primitive_dict` function to recursively filter out invalid python types in nested lists and dictionaries. This ensures that only valid python types are present in the inputs object before serialization for storage in the database.

### CODE REVIEW

File: src/backend/base/langflow/graph/utils.py [hunk 1]
```diff
@@ -125,14 +125,25 @@ def post_process_raw(raw, artifact_type: str):
...
+            # if it is a list we need to check if the contents are python types
+            filtered_list = [filter_valid_python_elements(v) for v in value]
+            return [v for v in filtered_list if v is not None]
+        if isinstance(value, dict):
+            # if it is a dict we need to check if the values are python types
+            filtered_dict = {k: filter_valid_python_elements(v) for k, v in value.items()}
+            return {k: v for k, v in filtered_dict.items() if v is not None}
+        if isinstance(value, str | int | bool | float):
+            return value
+        return None
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/graph/utils.py:140]: Here you could try to get something like `value.__name__` or `value.__class__.__name__` depending on what type of object it is

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]
Hey @dhlidongming 

Thanks for this

File: src/backend/base/langflow/graph/utils.py (hunk 1 above)
Comment (dhlidongming) [reply src/backend/base/langflow/graph/utils.py:144]: @ogabrielluiz Thank you for the suggestion! I've updated the code to handle `Document`, `Message`, and `Data` types appropriately. 
Could you please review the changes and let me know if this approach looks good to you?

**Comment by dhlidongming** [comment]
I've found that in the latest version, serialization to the database doesn't error out even if `inputs` contain non-Python objects. So, I'm closing this pull request.

**Pull request closed without merging**